### PR TITLE
Fix plymouth on multi-screens with different resolution

### PR DESCRIPTION
**Pull request by alimirjamali**
fixes: https://gitlab.freedesktop.org/plymouth/plymouth/-/issues/205

Follow-up to this discussion:
https://github.com/QubesOS/qubes-artwork/pull/40#issuecomment-2212593484

### CODE REVIEW

**Comment by alimirjamali** [comment]
@marmarek I do not have a PiKVM to test the above fix on it.

It fixes the issue on my own setup but the external display has higher resolution than the internal in my setup.

File: plymouth/qubes-dark/qubes-dark.script [hunk 1]
```diff
@@ -6,9 +6,19 @@ logo.sprite = Sprite(logo.image);
 logo.opacity_angle = 0;
 
 fun refresh_callback () {
+    # Reposition everything 50 times per second
```
Comment (marmarek) [inline plymouth/qubes-dark/qubes-dark.script:9]: This looks scary ;) But if it helps, I'm happy :)

Comment (alimirjamali) [reply plymouth/qubes-dark/qubes-dark.script:9]: > This looks scary ;) But if it helps, I'm happy :)

I am not 100% sure if it fixes it. Actually the logo is repositioned every 50 seconds at the moment with the existing code before this fix. I am guessing that the plymouth render interleaves the default 50 refresh cycles between the two monitors. But I might be wrong.

Comment (alimirjamali) [reply plymouth/qubes-dark/qubes-dark.script:9]: So it was a failure:

https://openqa.qubes-os.org/tests/106830#step/firstboot/3

Comment (marmarek) [reply plymouth/qubes-dark/qubes-dark.script:9]: Not yet, this step doesn't have the update installed yet.

Comment (marmarek) [reply plymouth/qubes-dark/qubes-dark.script:9]: This one is with the update applied: https://openqa.qubes-os.org/tests/106932#step/startup/4
So, not looking good. When the current run completes, I'll double check if initramfs actually include your change.

Comment (marmarek) [reply plymouth/qubes-dark/qubes-dark.script:9]: I have checked and your change is included there, and didn't help :(

Comment (marmarek) [reply plymouth/qubes-dark/qubes-dark.script:9]: In the plymouth issue I included a debug log, but also info in which plymouth version it worked before. Maybe that will help...

**Comment by marmarek** [comment]
It's even funnier in plymouth from F41: https://openqa.qubes-os.org/tests/109762#step/firstboot/2, https://openqa.qubes-os.org/tests/109764#step/firstboot/14 but maybe a bit better?

**Comment by alimirjamali** [comment]
> It's even funnier in plymouth from F41: https://openqa.qubes-os.org/tests/109762#step/firstboot/2, https://openqa.qubes-os.org/tests/109764#step/firstboot/14 but maybe a bit better?

It makes very sense. The patch I submitted only recenters the dialog and the white box. Not the logo. I will add the logo shortly. Hopefully tonight.

**Comment by marmarek** [comment]
Note the linked screenshots do not use this PR

**Comment by alimirjamali** [comment]
The problem with my local tests is that the HDMI capture dongle which I use as the external monitor emulates a Full-HD monitor which has higher resolution than the laptop itself. So it is very hard  to reproduce the bug. I have not found a way to force lower resolution on it.

Are you using PyKVM as the external monitor?

**Comment by marmarek** [comment]
> Are you using PyKVM as the external monitor?

In this case yes, but I can reproduce the issue also with a FullHD monitor, while the internal panel is 4K (or 3K? some high resolution).

Maybe there is some kernel/plymouth cmdline option to force specific resolution?

**Comment by marmarek** [comment]
And then, after pressing esc twice, the password entry migrates to the side, below the logo...

**Comment by marmarek** [comment]
The problem was elsewhere: https://github.com/QubesOS/qubes-artwork/pull/44

**Pull request closed without merging**